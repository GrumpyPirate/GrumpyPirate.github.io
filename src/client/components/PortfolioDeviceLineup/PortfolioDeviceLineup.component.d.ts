interface PortfolioDeviceLineupProps {
  desktopImage: string;
  tabletImage: string;
  mobileImage: string;
}

export { PortfolioDeviceLineupProps };
