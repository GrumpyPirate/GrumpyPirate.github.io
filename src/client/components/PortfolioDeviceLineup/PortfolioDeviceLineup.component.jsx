import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { uniqueId } from 'lodash-es';

import frameMobile from 'images/device-frames/frame-mobile.svg';
import frameTablet from 'images/device-frames/frame-tablet.svg';
import frameLaptop from 'images/device-frames/frame-laptop.svg';

import classes from './PortfolioDeviceLineup.component.scss';

const PortfolioDeviceLineup = ({ desktopImage, tabletImage, mobileImage }) => {
  const items = [
    {
      key: `portfolio-device-lineup__item--${uniqueId()}`,
      type: 'desktop',
      frameSrc: frameLaptop,
      imageSrc: desktopImage,
    },
    {
      key: `portfolio-device-lineup__item--${uniqueId()}`,
      type: 'tablet',
      frameSrc: frameTablet,
      imageSrc: tabletImage,
    },
    {
      key: `portfolio-device-lineup__item--${uniqueId()}`,
      type: 'mobile',
      frameSrc: frameMobile,
      imageSrc: mobileImage,
    },
  ];

  return (
    <div className={classes['portfolio-device-lineup']}>
      {items.map(item => (
        <div
          key={item.key}
          className={classnames(
            classes['portfolio-device-lineup__item'],
            classes[`portfolio-device-lineup__item--${item.type}`],
          )}
        >
          <figure className={classes['portfolio-device-lineup__figure']}>
            <img
              src={item.frameSrc}
              alt=""
              className={classes['portfolio-device-lineup__device-frame']}
            />
            <div className={classes['portfolio-device-lineup__device-img']}>
              <img src={item.imageSrc} alt="" />
            </div>
          </figure>
        </div>
      ))}
    </div>
  );
};

PortfolioDeviceLineup.propTypes = {
  desktopImage: PropTypes.string.isRequired,
  tabletImage: PropTypes.string.isRequired,
  mobileImage: PropTypes.string.isRequired,
};

export default PortfolioDeviceLineup;
