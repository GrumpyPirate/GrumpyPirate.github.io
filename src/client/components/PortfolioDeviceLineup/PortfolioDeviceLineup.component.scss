@import 'core';

$selector: portfolio-device-lineup;

.#{$selector} {
  height: rem(360px);
  position: relative;

  @include respond-up('xl') {
    height: rem(420px);
  }

  @include respond-down('sm') {
    display: none;
  }

  &__item {
    left: 50%;
    position: absolute;

    &--mobile {
      $aspect: (439 / 222);
      $device-frame-width-md: 96;
      $device-frame-width-xl: $device-frame-width-md * 1.2;

      bottom: 0;
      transform: translateX(75%);

      .#{$selector}__figure {
        height: rem($device-frame-width-md * $aspect * 1px);
        width: rem($device-frame-width-md * 1px);

        @include respond-up('xl') {
          height: rem($device-frame-width-xl * $aspect * 1px);
          width: rem($device-frame-width-xl * 1px);
        }
      }

      .#{$selector}__device-img {
        height: 74.75%;
        left: 6.8%;
        top: 13.125%;
        width: 86.25%;
      }
    }

    &--tablet {
      $aspect: (477 / 322);
      $device-frame-width-md: 180;
      $device-frame-width-xl: $device-frame-width-md * 1.2;

      bottom: 10px;
      transform: translateX(66%);

      .#{$selector}__figure {
        height: rem($device-frame-width-md * $aspect * 1px);
        width: rem($device-frame-width-md * 1px);

        @include respond-up('xl') {
          height: rem($device-frame-width-xl * $aspect * 1px);
          width: rem($device-frame-width-xl * 1px);
        }
      }

      .#{$selector}__device-img {
        height: 79.85%;
        left: 5.75%;
        top: 9.4%;
        width: 88.95%;
      }
    }

    &--desktop {
      $aspect: (291 / 506);
      $device-frame-width-md: 540;
      $device-frame-width-xl: $device-frame-width-md * 1.2;

      bottom: 20px;
      transform: translateX(-50%);

      .#{$selector}__figure {
        height: rem($device-frame-width-md * $aspect * 1px);
        width: rem($device-frame-width-md * 1px);

        @include respond-up('xl') {
          height: rem($device-frame-width-xl * $aspect * 1px);
          width: rem($device-frame-width-xl * 1px);
        }
      }

      .#{$selector}__device-img {
        height: 82%;
        left: 12.4%;
        top: 6%;
        width: 75.125%;
      }
    }
  }

  &__figure {
    margin: 0;
    position: relative;
  }

  &__device-frame {
    display: block;
    height: 100%;
    left: 0;
    pointer-events: none;
    position: absolute;
    top: 0;
    width: 100%;
    z-index: 2;
  }

  &__device-img {
    display: block;
    margin: 0;
    overflow-y: auto;
    position: relative;
    z-index: 1;

    img {
      display: block;
      height: auto;
      width: 100%;
    }
  }
}
