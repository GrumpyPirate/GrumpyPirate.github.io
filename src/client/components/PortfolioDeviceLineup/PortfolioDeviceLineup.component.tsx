import React, { FunctionComponent } from 'react';
import classnames from 'classnames';

import frameMobile from 'images/device-frames/frame-mobile.svg';
import frameTablet from 'images/device-frames/frame-tablet.svg';
import frameLaptop from 'images/device-frames/frame-laptop.svg';

import { PortfolioDeviceLineupProps } from './PortfolioDeviceLineup.component.d';
import classes from './PortfolioDeviceLineup.component.scss';

interface LineupItem {
  key: number;
  type: string;
  frameSrc: string;
  imageSrc: string;
}

const PortfolioDeviceLineup: FunctionComponent<PortfolioDeviceLineupProps> = ({
  desktopImage,
  tabletImage,
  mobileImage,
}: PortfolioDeviceLineupProps) => {
  const items: LineupItem[] = [
    {
      key: 1,
      type: 'desktop',
      frameSrc: frameLaptop,
      imageSrc: desktopImage,
    },
    {
      key: 2,
      type: 'tablet',
      frameSrc: frameTablet,
      imageSrc: tabletImage,
    },
    {
      key: 3,
      type: 'mobile',
      frameSrc: frameMobile,
      imageSrc: mobileImage,
    },
  ];

  return (
    <div className={classes['portfolio-device-lineup']}>
      {items.map(({ frameSrc, imageSrc, key, type }: LineupItem) => (
        <div
          key={key}
          className={classnames(
            classes['portfolio-device-lineup__item'],
            classes[`portfolio-device-lineup__item--${type}`],
          )}
        >
          <figure className={classes['portfolio-device-lineup__figure']}>
            <img
              src={frameSrc}
              alt=""
              className={classes['portfolio-device-lineup__device-frame']}
            />
            <div className={classes['portfolio-device-lineup__device-img']}>
              <img src={imageSrc} alt="" />
            </div>
          </figure>
        </div>
      ))}
    </div>
  );
};

export default PortfolioDeviceLineup;
export { LineupItem };
