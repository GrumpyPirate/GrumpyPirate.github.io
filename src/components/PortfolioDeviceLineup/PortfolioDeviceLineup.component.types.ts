export interface LineupItem {
  key: number;
  type: string;
  frameSrc: string;
  imageSrc: string;
}

export interface PortfolioDeviceLineupProps {
  desktopImage: string;
  tabletImage: string;
  mobileImage: string;
}
