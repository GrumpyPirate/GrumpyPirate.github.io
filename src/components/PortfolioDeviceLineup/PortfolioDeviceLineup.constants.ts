import styled, { css } from 'styled-components';

import { mediaQueries, rem } from 'styles';

import { LineupItemType } from './PortfolioDeviceLineup.types';

const mobileAspect = 439 / 222;
const mobileDeviceFrameWidthMd = 96;
const mobileDeviceFrameWidthXl = mobileDeviceFrameWidthMd * 1.2;

const tabletAspect = 477 / 322;
const tabletDeviceFrameWidthMd = 180;
const tabletDeviceFrameWidthXl = tabletDeviceFrameWidthMd * 1.2;

const desktopAspect = 291 / 506;
const desktopDeviceFrameWidthMd = 540;
const desktopDeviceFrameWidthXl = desktopDeviceFrameWidthMd * 1.2;

export const DevicePreview = styled.figure`
  margin: 0;
  position: relative;
`;

export const DeviceFrame = styled.img`
  display: block;
  height: 100%;
  left: 0;
  pointer-events: none;
  position: absolute;
  top: 0;
  width: 100%;
  z-index: 2;
`;

export const DeviceImage = styled.figure`
  display: block;
  margin: 0;
  overflow-y: auto;
  position: relative;
  z-index: 1;

  img {
    display: block;
    height: auto;
    width: 100%;
  }
`;

export const LineupItem = styled.div<{ type: LineupItemType }>`
  left: 50%;
  position: absolute;

  ${({ type }) => {
    switch (type) {
      case 'desktop':
        return css`
          bottom: 20px;
          transform: translateX(-50%);

          ${DevicePreview} {
            height: ${rem(desktopDeviceFrameWidthMd * desktopAspect)};
            width: ${rem(desktopDeviceFrameWidthMd)};

            @media ${mediaQueries.xl} {
              height: ${rem(desktopDeviceFrameWidthXl * desktopAspect)};
              width: ${rem(desktopDeviceFrameWidthXl)};
            }
          }

          ${DeviceImage} {
            height: 82%;
            left: 12.4%;
            top: 6%;
            width: 75.125%;
          }
        `;
      case 'tablet':
        return css`
          bottom: 10px;
          transform: translateX(66%);

          ${DevicePreview} {
            height: ${rem(tabletDeviceFrameWidthMd * tabletAspect)};
            width: ${rem(tabletDeviceFrameWidthMd)};

            @media ${mediaQueries.xl} {
              height: ${rem(tabletDeviceFrameWidthXl * tabletAspect)};
              width: ${rem(tabletDeviceFrameWidthXl)};
            }
          }

          ${DeviceImage} {
            height: 79.85%;
            left: 5.75%;
            top: 9.4%;
            width: 88.95%;
          }
        `;
      case 'mobile':
      default:
        return css`
          bottom: 0;
          transform: translateX(75%);

          ${DevicePreview} {
            height: ${rem(mobileDeviceFrameWidthMd * mobileAspect)};
            width: ${rem(mobileDeviceFrameWidthMd)};

            @media ${mediaQueries.xl} {
              height: ${rem(mobileDeviceFrameWidthXl * mobileAspect)};
              width: ${rem(mobileDeviceFrameWidthXl)};
            }
          }

          ${DeviceImage} {
            height: 74.75%;
            left: 6.8%;
            top: 13.125%;
            width: 86.25%;
          }
        `;
    }
  }}
`;
