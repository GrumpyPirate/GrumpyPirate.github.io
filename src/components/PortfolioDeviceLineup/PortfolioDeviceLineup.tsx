import React, { FunctionComponent } from 'react';
import styled from 'styled-components';

import LoadableImage from 'components/LoadableImage/LoadableImage';
import frameLaptop from 'images/device-frames/frame-laptop.svg';
import frameMobile from 'images/device-frames/frame-mobile.svg';
import frameTablet from 'images/device-frames/frame-tablet.svg';
import { ContentService } from 'services/ContentService';
import { mediaQueries, rem } from 'styles';

import {
  DeviceFrame,
  DeviceImage,
  DevicePreview,
  LineupItem,
} from './PortfolioDeviceLineup.constants';
import { LineupItemData, PortfolioDeviceLineupProps } from './PortfolioDeviceLineup.types';

const PortfolioDeviceLineup: FunctionComponent<PortfolioDeviceLineupProps> = ({
  className,
  desktopImage,
  tabletImage,
  mobileImage,
}) => {
  const items: LineupItemData[] = [
    {
      key: 1,
      type: 'desktop',
      frameSrc: frameLaptop,
      imageSrc: desktopImage,
    },
    {
      key: 2,
      type: 'tablet',
      frameSrc: frameTablet,
      imageSrc: tabletImage,
    },
    {
      key: 3,
      type: 'mobile',
      frameSrc: frameMobile,
      imageSrc: mobileImage,
    },
  ];

  return (
    <div className={className}>
      {items.map(({ frameSrc, imageSrc, key, type }) => (
        <LineupItem key={key} type={type}>
          <DevicePreview>
            <DeviceFrame src={frameSrc} alt="" />
            <DeviceImage>
              <picture>
                <source
                  type="image/webp"
                  srcSet={`${ContentService.getResizedImage(imageSrc, {
                    format: 'webp',
                    width: 486,
                  })} 486w, ${ContentService.getResizedImage(imageSrc, {
                    format: 'webp',
                    width: 972,
                  })} 972w`}
                  sizes="486px"
                />
                <source
                  type="image/jpeg"
                  srcSet={`${ContentService.getResizedImage(imageSrc, {
                    format: 'jpg',
                    width: 486,
                  })} 486w, ${ContentService.getResizedImage(imageSrc, {
                    format: 'jpg',
                    width: 972,
                  })} 972w`}
                  sizes="486px"
                />

                <source
                  type="image/webp"
                  srcSet={`${ContentService.getResizedImage(imageSrc, {
                    format: 'webp',
                    width: 406,
                  })} 406w, ${ContentService.getResizedImage(imageSrc, {
                    format: 'webp',
                    width: 812,
                  })} 812w`}
                  sizes="406px"
                />
                <source
                  type="image/jpeg"
                  srcSet={`${ContentService.getResizedImage(imageSrc, {
                    format: 'jpg',
                    width: 406,
                  })} 406w, ${ContentService.getResizedImage(imageSrc, {
                    format: 'jpg',
                    width: 812,
                  })} 812w`}
                  sizes="406px"
                />

                <LoadableImage loading="lazy" src={imageSrc} alt="" />
              </picture>
            </DeviceImage>
          </DevicePreview>
        </LineupItem>
      ))}
    </div>
  );
};

export default styled(PortfolioDeviceLineup)`
  height: ${rem(360)};
  position: relative;

  @media ${mediaQueries.xl} {
    height: ${rem(420)};
  }

  @media ${mediaQueries.smDown} {
    display: none;
  }
`;
