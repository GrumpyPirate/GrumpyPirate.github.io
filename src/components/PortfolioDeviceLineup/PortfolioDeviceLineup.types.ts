import { ClassNameProps } from 'types/common';

export type LineupItemType = 'desktop' | 'tablet' | 'mobile';

export interface LineupItemData {
  key: number;
  type: LineupItemType;
  frameSrc: string;
  imageSrc: string;
}

export interface PortfolioDeviceLineupProps extends ClassNameProps {
  desktopImage: string;
  tabletImage: string;
  mobileImage: string;
}
