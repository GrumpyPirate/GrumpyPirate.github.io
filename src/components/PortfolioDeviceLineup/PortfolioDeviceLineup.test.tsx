import { shallow } from 'enzyme';
import React from 'react';

import PortfolioDeviceLineup from './PortfolioDeviceLineup';
import fixture from './PortfolioDeviceLineup.fixture';

describe('Components', () => {
  describe('PortfolioDeviceLineup', () => {
    const props = {
      ...fixture,
    };

    it('should render as expected, without crashing', () => {
      const wrapper = shallow(<PortfolioDeviceLineup {...props} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
