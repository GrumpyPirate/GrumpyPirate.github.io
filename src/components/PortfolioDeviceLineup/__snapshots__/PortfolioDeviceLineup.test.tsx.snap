// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components PortfolioDeviceLineup should render as expected, without crashing 1`] = `
.c7 {
  opacity: 0;
  visibility: hidden;
  -webkit-transition-property: visibility,opacity;
  transition-property: visibility,opacity;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-timing-function: ease;
  transition-timing-function: ease;
}

.c3 {
  margin: 0;
  position: relative;
}

.c4 {
  display: block;
  height: 100%;
  left: 0;
  pointer-events: none;
  position: absolute;
  top: 0;
  width: 100%;
  z-index: 2;
}

.c6 {
  display: block;
  margin: 0;
  overflow-y: auto;
  position: relative;
  background-color: #333;
  z-index: 1;
}

.c6 img {
  display: block;
  height: auto;
  width: 100%;
}

.c1 {
  left: 50%;
  position: absolute;
  bottom: 20px;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c1 .c2 {
  height: 19.409584980237156rem;
  width: 33.75rem;
}

.c1 .c5 {
  height: 82%;
  left: 12.4%;
  top: 6%;
  width: 75.125%;
}

.c8 {
  left: 50%;
  position: absolute;
  bottom: 10px;
  -webkit-transform: translateX(66%);
  -ms-transform: translateX(66%);
  transform: translateX(66%);
}

.c8 .c2 {
  height: 16.665372670807454rem;
  width: 11.25rem;
}

.c8 .c5 {
  height: 79.85%;
  left: 5.75%;
  top: 9.4%;
  width: 88.95%;
}

.c9 {
  left: 50%;
  position: absolute;
  bottom: 0;
  -webkit-transform: translateX(75%);
  -ms-transform: translateX(75%);
  transform: translateX(75%);
}

.c9 .c2 {
  height: 11.864864864864865rem;
  width: 6rem;
}

.c9 .c5 {
  height: 74.75%;
  left: 6.8%;
  top: 13.125%;
  width: 86.25%;
}

.c0 {
  height: 22.5rem;
  position: relative;
}

@media (min-width:90em) {
  .c1 .c2 {
    height: 23.291501976284586rem;
    width: 40.5rem;
  }
}

@media (min-width:90em) {
  .c8 .c2 {
    height: 19.998447204968944rem;
    width: 13.5rem;
  }
}

@media (min-width:90em) {
  .c9 .c2 {
    height: 14.237837837837837rem;
    width: 7.199999999999999rem;
  }
}

@media (min-width:90em) {
  .c0 {
    height: 26.25rem;
  }
}

@media (max-width:47.9375em) {
  .c0 {
    display: none;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
    type="desktop"
  >
    <figure
      class="c2 c3"
    >
      <img
        alt=""
        class="c4"
        src="frame-laptop.svg"
      />
      <figure
        class="c5 c6"
      >
        <picture>
          <source
            sizes="486px"
            srcset="http://lobola.com/gonothecal/backiebird?a=waldsteinia&b=exorcistic#ketch?fm=webp&w=486 486w, http://lobola.com/gonothecal/backiebird?a=waldsteinia&b=exorcistic#ketch?fm=webp&w=972 972w"
            type="image/webp"
          />
          <source
            sizes="486px"
            srcset="http://lobola.com/gonothecal/backiebird?a=waldsteinia&b=exorcistic#ketch?fm=jpg&w=486 486w, http://lobola.com/gonothecal/backiebird?a=waldsteinia&b=exorcistic#ketch?fm=jpg&w=972 972w"
            type="image/jpeg"
          />
          <source
            sizes="406px"
            srcset="http://lobola.com/gonothecal/backiebird?a=waldsteinia&b=exorcistic#ketch?fm=webp&w=406 406w, http://lobola.com/gonothecal/backiebird?a=waldsteinia&b=exorcistic#ketch?fm=webp&w=812 812w"
            type="image/webp"
          />
          <source
            sizes="406px"
            srcset="http://lobola.com/gonothecal/backiebird?a=waldsteinia&b=exorcistic#ketch?fm=jpg&w=406 406w, http://lobola.com/gonothecal/backiebird?a=waldsteinia&b=exorcistic#ketch?fm=jpg&w=812 812w"
            type="image/jpeg"
          />
          <img
            alt=""
            class="c7"
            data-testid="loadable-image"
            loading="lazy"
            src="http://lobola.com/gonothecal/backiebird?a=waldsteinia&b=exorcistic#ketch"
          />
        </picture>
      </figure>
    </figure>
  </div>
  <div
    class="c8"
    type="tablet"
  >
    <figure
      class="c2 c3"
    >
      <img
        alt=""
        class="c4"
        src="frame-tablet.svg"
      />
      <figure
        class="c5 c6"
      >
        <picture>
          <source
            sizes="486px"
            srcset="http://sanguineous.com/subquadrangular/pararosaniline?a=baalize&b=scaramouch#unsickly?fm=webp&w=486 486w, http://sanguineous.com/subquadrangular/pararosaniline?a=baalize&b=scaramouch#unsickly?fm=webp&w=972 972w"
            type="image/webp"
          />
          <source
            sizes="486px"
            srcset="http://sanguineous.com/subquadrangular/pararosaniline?a=baalize&b=scaramouch#unsickly?fm=jpg&w=486 486w, http://sanguineous.com/subquadrangular/pararosaniline?a=baalize&b=scaramouch#unsickly?fm=jpg&w=972 972w"
            type="image/jpeg"
          />
          <source
            sizes="406px"
            srcset="http://sanguineous.com/subquadrangular/pararosaniline?a=baalize&b=scaramouch#unsickly?fm=webp&w=406 406w, http://sanguineous.com/subquadrangular/pararosaniline?a=baalize&b=scaramouch#unsickly?fm=webp&w=812 812w"
            type="image/webp"
          />
          <source
            sizes="406px"
            srcset="http://sanguineous.com/subquadrangular/pararosaniline?a=baalize&b=scaramouch#unsickly?fm=jpg&w=406 406w, http://sanguineous.com/subquadrangular/pararosaniline?a=baalize&b=scaramouch#unsickly?fm=jpg&w=812 812w"
            type="image/jpeg"
          />
          <img
            alt=""
            class="c7"
            data-testid="loadable-image"
            loading="lazy"
            src="http://sanguineous.com/subquadrangular/pararosaniline?a=baalize&b=scaramouch#unsickly"
          />
        </picture>
      </figure>
    </figure>
  </div>
  <div
    class="c9"
    type="mobile"
  >
    <figure
      class="c2 c3"
    >
      <img
        alt=""
        class="c4"
        src="frame-mobile.svg"
      />
      <figure
        class="c5 c6"
      >
        <picture>
          <source
            sizes="486px"
            srcset="https://inconcinnately.com/amputate/nebelist?a=bepill&b=smokelessness#vaulting?fm=webp&w=486 486w, https://inconcinnately.com/amputate/nebelist?a=bepill&b=smokelessness#vaulting?fm=webp&w=972 972w"
            type="image/webp"
          />
          <source
            sizes="486px"
            srcset="https://inconcinnately.com/amputate/nebelist?a=bepill&b=smokelessness#vaulting?fm=jpg&w=486 486w, https://inconcinnately.com/amputate/nebelist?a=bepill&b=smokelessness#vaulting?fm=jpg&w=972 972w"
            type="image/jpeg"
          />
          <source
            sizes="406px"
            srcset="https://inconcinnately.com/amputate/nebelist?a=bepill&b=smokelessness#vaulting?fm=webp&w=406 406w, https://inconcinnately.com/amputate/nebelist?a=bepill&b=smokelessness#vaulting?fm=webp&w=812 812w"
            type="image/webp"
          />
          <source
            sizes="406px"
            srcset="https://inconcinnately.com/amputate/nebelist?a=bepill&b=smokelessness#vaulting?fm=jpg&w=406 406w, https://inconcinnately.com/amputate/nebelist?a=bepill&b=smokelessness#vaulting?fm=jpg&w=812 812w"
            type="image/jpeg"
          />
          <img
            alt=""
            class="c7"
            data-testid="loadable-image"
            loading="lazy"
            src="https://inconcinnately.com/amputate/nebelist?a=bepill&b=smokelessness#vaulting"
          />
        </picture>
      </figure>
    </figure>
  </div>
</div>
`;
